python/ray/autoscaler/_private/event_system.py

```diff
@@ -38,6 +51,64 @@ class CreateClusterEvent(Enum):
     cluster_booting_completed = auto()
 
 
+class ScriptStartedEvent(Enum):
+    """Events to track for Ray scripts that are executed.
+    """
+    @property
+    def state(self) -> str:
+        return States.STARTED.name
+
+    start_initializing = auto()
+
+
+class ScriptInProgressEvent(Enum):
+    """Events to track during execution of Ray scripts.
+    """
+    @property
+    def state(self) -> str:
+        return States.IN_PROGRESS.name
+
+    in_progress = auto()
+
+
+class ScriptInProgressCustomEvent:
+    """Custom, user-defined events to track during execution of Ray scripts.
+    """
+    @property
+    def state(self) -> str:
+        return States.IN_PROGRESS.name
```

Comment:
Should we just let the user define the parent state that they're annotating with additional state sequence information during construction, and then change this class name to something like `SequentialEvent`,`EventSequence`?

Reply:
I think that makes sense, in the spirit of making the event system extensible to users (for our use case, defining event states in DeltaCAT instead).